classes/robot/crawler.php

```diff
@@ -902,7 +902,7 @@ private function link_from_node_to_url($from, $url, $text, $idattr) {
         global $DB;
 
         // Add the node URL to the queue.
-        $to = $this->mark_for_crawl($from->url, $url);
+        $to = $this->mark_for_crawl($from->url, $url, null, $from->priority);
```

Comment:
So lets say I mark A as high priority

A then links to, B,C,D,E,F

By the looks of it, this will mean all B-F will be high priority. And then when we scrape those, every link on those pages will end up high priority.And so on. I haven't tested this but it looks like priority will infect everything. But we only want it to affect this page, and the links on this page.

ie I think conceptually we should be 'this node' as high priority, as well as marking the outgoing 'edge's of this node as high priority, not the nodes.

Implementation wise, I'm not sure we actually need to add a priority column to the edge table, an easier solution might be to have 2 levels of priority: one which inherits and one which doesn't. We mark the starting node as priority 2 = inherit, and then if it's 2 then we mark it's children as 1